### ISSUE TITLE
[Home Assistant] Cannot include any other device after update of zwavejs2mqtt

### ISSUE THREAD

**Issue by giovi321** [zwave-js/zwavejs2mqtt#1949]
### Version

Build/Run method:

[x] Docker
[ ] Git Clone + yarn install + yarn run build + yarn start
[ ] Pkg

Zwavejs2Mqtt version: **5.10.1.91ef490**
Z-Wave JS version: **8.6.1**
### Describe the bug

Information that could be useful:
- running on raspberry pi3b+
- home assistant on venv 2021.9.3

Whether I try to include a new device to the network (both in secure and insecure ways, both from home assistant and from zwavejs2mqtt control panel) I get no inclusion and all I get is the screen "Inclusion is started. Please put your device in INCLUSION MODE" stuck at 1 second left.
I tried two different devices:
1) Danfoss LC13 - which was successfully added in the past before the update of zwavejs2mqtt
2) AEON Labs Multisensor 6 - of which there are already 3 included in the network

Something odd I noticed: some filesystems seem to be 100% used (not root)
```
root@hass:~# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        55G   34G   19G  64% /
devtmpfs        1.8G     0  1.8G   0% /dev
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           1.9G  210M  1.7G  11% /run
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs           1.9G     0  1.9G   0% /sys/fs/cgroup
/dev/loop1       52M   52M     0 100% /snap/core20/1170
/dev/loop2      145M  145M     0 100% /snap/zwavejs2mqtt/239
/dev/loop0       85M   85M     0 100% /snap/core/11997
/dev/loop3       85M   85M     0 100% /snap/core/11610
/dev/loop4      117M  117M     0 100% /snap/zwavejs2mqtt/253
/dev/loop5       52M   52M     0 100% /snap/core20/1085
/dev/mmcblk0p1  253M   49M  204M  20% /boot
tmpfs           384M     0  384M   0% /run/user/0

```

[zwave_js_dump.json.txt](https://github.com/zwave-js/zwavejs2mqtt/files/7500647/zwave_js_dump.json.txt)

**Comment by robertsLando** [comment]
Are you using snap?

**Comment by giovi321** [comment]
@robertsLando yes, I am

**Comment by robertsLando** [comment]
Could be something related to it, I mean about the disk usage. Could you attach some logs please? You have attached nodes bump but it's not useful here.

cc @jmgiaever

**Comment by giovi321** [comment]
Logs such as? Syslog?

**Comment by jmgiaever** [comment]
Hi, no. Those "devices" is the read only filesystems that the system files of the snap lives in. It doesn't make sense that these filesystems should have any space left since they are RO. 

The RW filesystem should be something /var/snap/zwavejs2mqtt/<revision number> while the RO is /snap/zwavejs2mqtt/<revision number>. 

But I don't understand. Do you use docker or the snap package or z2m? The issue says docker but the output of `df` shows the snap version is there too.

**Comment by jmgiaever** [comment]
PS! If you're running the snap package, could you give the release in the channel `latest/candidate` a try

```
sudo snap switch zwavejs2mqtt --channel=latest/candidate
sudo snap refresh zwavejs2mqtt
```

The security bootstrapping seems much faster within this release.

**Comment by giovi321** [comment]
@robertsLando, please find attached the logs as requested
[zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7503735/zwavejs2mqtt-store.zip)

@jmgiaever to be completely honest I was sure I installed through snap, but now you're making me doubt it. 
I have tried:
```
sudo snap switch zwavejs2mqtt --channel=latest/candidate
sudo snap refresh zwavejs2mqtt
```
and this is the output:
```
zwavejs2mqtt (candidate) v5.11.0 from Giaever.online (giaever-online) refreshed
```
Should I maybe reboot?

**Comment by robertsLando** [comment]
There is a ton of 

`2021-11-09T09:49:39.089Z DRIVER   Dropping message because the driver is not ready to handle it yet.`

In zwavejs logs. Dunno what's happening here as logs seems to be inclomplete (there si no startup logs). Could you try to drop your logs, restart application and try again to add a node? 

cc @AlCalzone

**Comment by giovi321** [comment]
Here is the log after a complete reboot of the machine and one try of adding a device
[zwavejs2mqtt-store1.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7503847/zwavejs2mqtt-store1.zip)
.

**Comment by jmgiaever** [comment]
> But nothing changes. Should I maybe reboot?
Can you give me the versions you're at now?

**Comment by giovi321** [comment]
@jmgiaever 
zwavejs2mqtt: 5.11.0.474d303
zwave-js: 8.7.5
home id: 3623972245
home hex: 0xd8016d95

**Comment by AlCalzone** [comment]
Please set the loglevel to "debug". The interesting parts are missing.

**Comment by jmgiaever** [comment]
That is latest. I think @AlCalzone must do the digging of the logs.

**Comment by giovi321** [comment]
> That is latest. I think @AlCalzone must do the digging of the logs.

Let me know if this is what you expected, and thank you for your help.
[zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7504317/zwavejs2mqtt-store.zip)

**Comment by AlCalzone** [comment]
The serial logs are still missing for some reason. Are you sure that's loglevel debug for the driver?

**Comment by giovi321** [comment]
@AlCalzone sorry, for some reason I didn't apply the new log configuration. Here's the log level "silly".
Thank you if you could have a look at it :)
[zwavejs2mqtt-store(1).zip](https://github.com/zwave-js/zwavejs2mqtt/files/7511081/zwavejs2mqtt-store.1.zip)

**Comment by AlCalzone** [comment]
I can't see anything definite why the inclusion doesn't work, but my money is on the flood of messages bringing the network to its knees.

Nodes 22-24 send all their measurements (temperature, humidity, brightness, battery, UV) every 2 seconds or so.
Node 11 isn't even known to the controller but still sends its measurements happily every couple of seconds.

Most likely the communication necessary for inclusion doesn't get any "air time" here. Turn down the reporting massively (timed reports roughly every 30 minutes if you even need them at all, prefer reporting on change) and I would assume that things go back to normal quickly.

**Comment by giovi321** [comment]
Thank you @AlCalzone I will give it a try. It's very weird htough because it has been working for over a year with this set-up

**Comment by giovi321** [comment]
No success :( I tried to reduce the measurements frequency but still no success... and how is it possible that there's a node 11 that is not even in the table?
What would you recommend me to do?

**Comment by robertsLando** [comment]
Dunno how many devices you have but IMO ecluding all them, hard reset stick and add them back one by one would be the cleanest solution

**Comment by giovi321** [comment]
I forgot to mention that when I add a device I get the following message: 
`Some security keys are missing: S2_Unauthenticated, S2_Authenticated, S2_AccessControl, S0_Legacy. Please check your zwave settings.`

**Comment by giovi321** [comment]
I have merely 5 devices + usb stick

**Comment by robertsLando** [comment]
Go to settings and add missing keys please, zwave section :)

**Comment by AlCalzone** [comment]
Please post the entire log of the inclusion process. Keys are only needed after the actual inclusion so this can't be the issue.

**Comment by giovi321** [comment]
Ok, I just excluded and re-included a working node and everything went smoothly. I'm starting to think that the two nodes I can't include are too far from the receiver but this surprises me because they are zwave+ so they can work as a "hop" like in my previous network graph (that now has changed with no "hops").
Concerning node 11, I have no clue who might it be!

**Comment by giovi321** [comment]
Ok it's not the distance. I moved one of the two devices that I'm trying to include closer to the zave stick and made no difference. I'm officially desperate.

**Comment by jmgiaever** [comment]
Hi. Did you do a HARD RESET of the stick, and re-included the devices? 

It sounds like there's something fishy going on since you have a node 11 that neither you or the controller knows about.

**Comment by giovi321** [comment]
I have performed a hard reset and I tried to add a factory reset device to the network (that was working till few days ago regularly) and this is the log I get. Any clue?

```

2021-11-10 11:07:12.246 INFO ZWAVE: Connecting to /dev/ttyACM0
2021-11-10 11:07:12.235 INFO MQTT: MQTT is disabled
2021-11-10 11:07:12.265 INFO APP: POST /api/settings 200 116.367 ms - 995
2021-11-10 11:07:12.295 INFO APP: GET /api/auth-enabled 304 2.106 ms - -
2021-11-10 11:07:21.339 INFO GATEWAY: Driver is READY
2021-11-10 11:07:21.341 INFO ZWAVE: Zwave driver is ready
2021-11-10 11:07:21.342 INFO ZWAVE: Controller status: Driver ready
2021-11-10 11:07:21.349 DEBUG ZWAVE: Binding to node 1 events
2021-11-10 11:07:21.352 DEBUG ZWAVE: Node 1 has been added to nodes array
2021-11-10 11:07:21.354 INFO ZWAVE: Scanning network with homeid: 0xed534109
2021-11-10 11:07:21.356 INFO ZWAVE: Node 1 is alive
2021-11-10 11:07:21.359 DEBUG ZWAVE: Node 1: mapping absence of CC 128 (Battery) to node property 'isBatteryPowered
2021-11-10 11:07:21.360 INFO ZWAVE: Node 1 ready: Sigma Designs (Former Zensys) - HUSBZB-1 (QuickStick Combo)
2021-11-10 11:07:21.364 INFO ZWAVE: Controller status: Scan completed
2021-11-10 11:07:21.365 INFO ZWAVE: Network scan complete. Found: 1 nodes
2021-11-10 11:07:21.646 INFO ZWAVE: Next update scheduled for: Thu Nov 11 2021 00:00:00 GMT+0000 (Greenwich Mean Time)
2021-11-10 11:07:31.359 DEBUG SOCKET: Event ZWAVE_API emitted to d7uiXpJRDiB6jEFiAAAD
2021-11-10 11:07:31.364 INFO ZWAVE: Calling api startInclusion with args: [ 0, { forceSecurity: false }, [length]: 2 ]
2021-11-10 11:07:31.407 INFO ZWAVE: Controller status: Secure inclusion started
2021-11-10 11:07:31.410 INFO ZWAVE: Success zwave api call startInclusion true
2021-11-10 11:08:01.469 INFO ZWAVE: Controller status: Inclusion stopped
2021-11-10 11:08:11.623 INFO APP: GET /api/store 200 13.535 ms - 2846
2021-11-10 11:08:21.033 DEBUG APP: zip archive ready
2021-11-10 11:08:21.036 INFO APP: POST /api/store-multi 200 50.492 ms - -
```

**Comment by AlCalzone** [comment]
Driver logs please

**Comment by giovi321** [comment]
@AlCalzone yes sorry made a mistake :)

**Comment by AlCalzone** [comment]
The inclusion process is stopped after 30 seconds without any message reaching the controller at all. To be sure, let's exclude a common issue: controller placement.

Z-Wave sticks in particular are prone to interference by USB ports, especially by USB3 ports. We recommend putting the stick in a suitable location:

-   on an **USB extension cord** (this works wonders!)
-   away from other USB ports
-   away from metallic surfaces
-   and especially not in the back of a server rack

**Comment by giovi321** [comment]
I tried the USB extension cord and I even placed the device 50cm away from the usb stick - no success.
What other option do I have?

**Comment by giovi321** [comment]
Ok I tried hard reset, switching cable, now it's not even including nodes that were working 60 minutes ago... maybe a reinstall could be helpful?

**Comment by jmgiaever** [comment]
Try to do a re-install;

```
sudo snap remove --purge zwavejs2mqtt
sudo snap install zwavejs2mqtt --channel=latest/candidate
sudo snap set zwavejs2mqtt server.host=0.0.0.0

sudo snap connect zwavejs2mqtt:raw-usb
sudo snap connect zwavejs2mqtt:hardware-observe

sudo zwavejs2mqtt.exec
```

When you see that everything runs fine, exit process and demonize it with

```
sudo zwavejs2mqtt.enable
```

**Comment by giovi321** [comment]
@jmgiaever thank you for the instructions.
I could wun with `/snap/bin/zwavejs2mqtt.exec` but i could not demonize with `zwavejs2mqtt.enable` nor with `/snap/bin/zwavejs2mqtt.enable`

**Comment by giovi321** [comment]
Ok I managed to reinstall with the specific procedure for Snap package for Raspberry PI. Still not getting any luck on adding most of the devices (mostly the Danfoss LC13 and, especially, the AEOTEC multisensor 6.

**Comment by AlCalzone** [comment]
Do you have a spare Z-Wave stick that you could flash with the Zniffer firmware to see what's actually happening on the radio?

**Comment by giovi321** [comment]
I don't know how, why, what have I done but now everything is back to normal and works. Installed through the specific procedure for Snap package for Raspberry PI and worked out of the box. I don't have the courage to update even if it shows the icon 
![image](https://user-images.githubusercontent.com/6443515/141148497-560d9565-3f37-4f07-9afe-e8f794d97687.png)

**Comment by jmgiaever** [comment]
I'm sure it's fine to do the update. That's just configuration files. 

But out of curiosity; which stick do you use?

**Comment by giovi321** [comment]
>But out of curiosity; which stick do you use? 
Everspring Z-Wave (on amazon [B01LQZ9BLH](https://www.amazon.it/dp/B01LQZ9BLG))

I updated and I'm having troubles again... some thermostatic valves are loosing their capabilities and the driver keeps going offline, they show "Unknown manufacturer 0xXXXX", another device shows capabilities on ZwaveJS2mqrtt but not on home assistant.

**Comment by jmgiaever** [comment]
Could you try to do a restart of the app?

`sudo snap restart zwavejs2mqtt`

**Comment by AlCalzone** [comment]
@giovi321 unfortunately your log includes neither a restart nor the config update, so it's unclear what is happening there.

**Comment by giovi321** [comment]
Sure, here it is.
[zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7519137/zwavejs2mqtt-store.zip)

**Comment by jmgiaever** [comment]
Did you try a simple restart? 

And can you share your settings file? Remove your network keys first. 

+ The output from `sudo snap info zwavejs2mqtt`

**Comment by giovi321** [comment]
I tried a simple restart and reboot, no change.

[settings.json.txt](https://github.com/zwave-js/zwavejs2mqtt/files/7519227/settings.json.txt)

```
root@hass:~# sudo snap info zwavejs2mqtt
name:      zwavejs2mqtt
summary:   Fully configurable Zwave to MQTT Gateway and Control Panel.
publisher: Giaever.online (giaever-online)
store-url: https://snapcraft.io/zwavejs2mqtt
license:   MIT
description: |
  **Fully configurable Zwave to MQTT Gateway and Control Panel**
  
  - **Zwave to Mqtt Gateway**: Configure how nodes and nodes values are mapped between the two
  protocols
  - **Secured**: Supports HTTPS and users authentication
  - **Control Panel UI**: Directly control all your nodes and their values from the UI, some of
  control panel features:
    - **Nodes management**: check all nodes discovered in the z-wave network and manage their values
    - **Firmware updates**: update devices firmware, just select the controller action Begin
    Firmware Update and upload the firmware file
    - **Groups associations**: add/edit nodes associations
    - Full zwave-js APIs support
  - **Custom scenes management**: create scenes and trigger them by using MQTT apis (also supports
  timeouts)
  - **Log debug in UI**: See debug logs directly from the UI
  - **Store directory management**: all files are stored in store folder, you can download/edit
  files inside this folder directly from the UI
  - **Network graph**: see how nodes are communicating with the controller, useful for diagnostics
  purposes
  - **Home Assistant integration**: you can use official Zwavejs integration by enabling zwavejs
  server or use integrated MQTT discovery integration
  
  **Channels in `latest`**
  
  This is how we use the different of channels in the `latest` track,
  to give the user a chance to choose how often they want updates.
  
  - `latest/stable`: Latest version of the last *major* release. Will                               
  only update once per major release and most likely by the end of
    that release cycle. E.g when there is a new version `b.a.a` then
    version `a.b.c` will be will be available.
  
  - `latest/candidate`:  Latest minor or patch release of latest major
    release. E.g when version `a.b.c` changes to `a.b.d` or `a.c.a`, then
    version `a.b.d` or `a.c.a` will be available. Consider this rolling
    releases.
  - `latest/edge` and `latest/edge/dev`: Every build (including tests)
    might be pushed to these channels.
commands:
  - zwavejs2mqtt.disable
  - zwavejs2mqtt.enable
  - zwavejs2mqtt.exec
  - zwavejs2mqtt.help
  - zwavejs2mqtt.restart
services:
  zwavejs2mqtt: simple, enabled, active
snap-id:      k26XpW9UYHL1HNh20isi5MHoazqHZ8br
tracking:     latest/stable
refresh-date: yesterday at 14:25 GMT
channels:
  latest/stable:    v5.10.1 2021-11-05 (253) 122MB -
  latest/candidate: v5.11.0 2021-11-05 (258) 129MB -
  latest/beta:      ^                              
  latest/edge:      v5.11.0 2021-11-05 (258) 129MB -
installed:          v5.10.1            (253) 122MB -
```

**Comment by AlCalzone** [comment]
Ok, installing the config update seems to be working okay. The restart at `2021-11-11T08:55:54.925Z` in combination with securely included multisensors then seem to overwhelm your stick.

We're back to my comment earlier https://github.com/zwave-js/zwavejs2mqtt/issues/1949#issuecomment-964916577, but even worse.

Node 3 and 5 are sending all their measurements (temperature, humidity, UV, luminance, battery) every 5s now, but also request nonces for secure transmission. This totals (2 sensors x 5 measurements x 3 messages per measurement) per 5s, meaning those 2 sensors alone are are responsible for 6 messages per second. This is way too much for a reliable Z-Wave network.

Every time the stick tries to send something while one of these messages gets received (and this happens often), there is a collision, it has to wait and try again later. These older firmwares sometimes get out of sync with the host in these cases too.

As a result, the controller's queries, either during interview or startup (seemingly) fail at random with the result you're getting.

TL;DR: 
* Turn the reporting intervals WAY up (the sensors don't need to remind the controller every 5 seconds that they are still at 51% Humidity) and make use of reporting on change.
* Disable the measurements you don't need (do you REALLY need UV for example?)
* Don't include multisensors with S0 unless you absolutely have to.

**Comment by giovi321** [comment]
@AlCalzone @jmgiaever @robertsLando I wish I could offer you at least a beer (let me know if you have a patron or something). 
Thank you so much for supporting me in problem solving z-wave, a technology I'm barely aware of.

@AlCalzone, do you think that getting a better (more expensive controller) could improve the situation?

Many many thanks again.

**Comment by robertsLando** [comment]
@giovi321 Sure, just press on sponsor button in our profile and you will see all the options available.

![Schermata da 2021-11-11 10-57-17](https://user-images.githubusercontent.com/11502495/141277935-0dfe97ef-c5f2-4aad-adf8-b7a402ba6c29.png)


My patron is: https://patreon.com/user?u=16906849
Or also: https://www.buymeacoffee.com/MVg9wc2HE

**Comment by AlCalzone** [comment]
@giovi321 maybe a new controller (not necessarily expensive, but the new 700 series if they are available) can help a bit. But the original problem will also be detrimental on a new network.
If you like to sponsor me, here you'll find the links (PayPal or Github Sponsors): https://github.com/zwave-js/node-zwave-js

**Comment by giovi321** [comment]
@AlCalzone I offered a beer, hope to be able to do it in person one day.
The 700 series what brand is it?

**Comment by robertsLando** [comment]
> The 700 series what brand is it

https://aeotec.com/z-wave-home-automation/z-wave-700-series.html

**Comment by giovi321** [comment]
I found it here if you're interested :)
https://www.fatelettronica.it/index.php?option=com_virtuemart&view=productdetails&virtuemart_product_id=877146&virtuemart_category_id=2756&Itemid=1&gclid=EAIaIQobChMI-oW1uKqQ9AIVweN3Ch1lRwWWEAQYAiABEgKEb_D_BwE

**Comment by robertsLando** [comment]
Yep I will get one soon, grazie Giovanni 🙏🏼 😄

**Comment by giovi321** [comment]
Least I could do!
